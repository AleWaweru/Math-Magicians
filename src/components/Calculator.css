.calculator-body {
  height: 19rem;
  width: 100%;
  margin-right: 5px;
  box-shadow: 5px 5px 8px 8px gray;
}

.calc-result {
  display: flex;
  text-align: right;
  background-color: rgb(155, 154, 154);
  max-width: 100%;
  height: 50px;
  color: #fff;
  border: 1px solid rgb(228, 225, 225);
  font-size: 30px;
  justify-content: flex-end;
  letter-spacing: 1px;
  padding-right: 5px;
}

.calc-result input::placeholder {
  color: white;
  padding-top: 10px;
  width: 100%;
}

.assign-operators {
  justify-content: space-between;
  width: 100%;
  display: block;
  height: 50px;
  background-color: rgb(247, 240, 240);
}

.first-row,
.second-row,
.third-row,
.fourth-row {
  display: flex;
  width: 100%;
  background-color: rgb(247, 240, 240);
  height: 50px;
}

.firth-row {
  display: flex;
  max-width: 100%;
  margin-top: 9.5rem;
  background-color: rgb(247, 240, 240);
  height: 50px;
}

.firth-row button {
  border: 1px solid rgb(228, 225, 225);
  width: 80%;
  padding-left: 80px;
  padding-top: 15px;
}

.assign-operators .first-row button {
  gap: 3rem;
  border: 1px solid rgb(228, 225, 225);
  width: 80%;
  padding-left: 80px;
  padding-top: 15px;
}

.assign-operators .second-row button {
  gap: 3rem;
  border: 1px solid rgb(228, 225, 225);
  width: 80%;
  padding-left: 80px;
  padding-top: 15px;
}

.assign-operators .third-row button {
  gap: 3rem;
  border: 1px solid rgb(228, 225, 225);
  width: 80%;
  padding-left: 80px;
  padding-top: 15px;
}

.assign-operators .fourth-row button {
  gap: 3rem;
  border: 1px solid rgb(228, 225, 225);
  width: 80%;
  padding-left: 80px;
  padding-top: 15px;
}

.special {
  background-color: rgb(240, 153, 121);
}

#division {
  font-size: 10px;
  font-weight: normal;
}

.firth-row .dot-width {
  max-width: 25%;
  font-size: 20px;
}

.firth-row .special {
  max-width: 25%;
}

.firth-row .zero-width {
  padding-left: 12rem;
}

.quotes-container {
  border: 1px solid grey;
  margin-top: 20px;
  box-shadow: 5px 5px 8px 8px gray;
}
