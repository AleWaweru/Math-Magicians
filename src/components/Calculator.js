import React from 'react';
import { FaDivide } from 'react-icons/fa';

import './Calculator.css';
import FirthRow from './calFirstRow';

const Calculator = () => (
  <div className="calculator-body">
    <div className="calc-result">
      <span>0</span>
    </div>
    <div className="assign-operators">
      <div className="first-row">
        <span>AC</span>
        <span>+/-</span>
        <span>%</span>
        <span className="special" id="division">
          <FaDivide />
        </span>
      </div>
      <div className="second-row">
        <span>7</span>
        <span>8</span>
        <span>9</span>
        <span className="special">x</span>
      </div>
      <div className="third-row">
        <span>4</span>
        <span>5</span>
        <span>6</span>
        <span className="special">-</span>
      </div>
      <div className="fourth-row">
        <span>1</span>
        <span>2</span>
        <span>3</span>
        <span className="special">+</span>
      </div>
    </div>
    <FirthRow />
  </div>
);

export default Calculator;
