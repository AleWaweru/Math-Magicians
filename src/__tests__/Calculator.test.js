import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Calculator', () => {
  test('renders the Calculator component correctly', () => {
    const component = renderer.create(<Calculator />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
