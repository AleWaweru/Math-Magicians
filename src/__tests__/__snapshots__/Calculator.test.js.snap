// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders the Calculator component correctly 1`] = `
<div
  className="calculator-body"
>
  <div
    className="calc-result"
    data-testid="result"
  />
  <div
    className="assign-operators"
  >
    <div
      className="first-row"
    >
      <button
        className="keys"
        id="clearAll"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="keys"
        id="backspace"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="keys"
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="keys special"
        id="division"
        name="/"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="second-row"
    >
      <button
        className="keys"
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="keys"
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="keys"
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="keys special"
        name="*"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="third-row"
    >
      <button
        className="keys"
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="keys"
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="keys"
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="keys special"
        name="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="fourth-row"
    >
      <button
        className="keys"
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="keys"
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="keys"
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="keys special"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
  </div>
  <div
    className="firth-row"
  >
    <button
      className="keys zero-width"
      name="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="keys dot-width"
      name="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="keys special"
      id="results"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
